marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow.mjs

```diff
@@ -122,17 +122,17 @@ export default class Flow {
     const prefix = flowName + "|" + fn.lowerCase(stepId) + "|finished|";
 
     // A map is used in this script to avoid N calls to array.includes
-    const script = "var items; var prefix; " +
-      "const itemHashes = items.map(item => xdmp.hash64(prefix + item)); " +
-      "const processedItemHashesMap = cts.values(cts.jsonPropertyReference('processedItemHashes'), null, ['map'], cts.andQuery([" +
-      "  cts.collectionQuery('Batch'), " +
-      "  cts.jsonPropertyRangeQuery('processedItemHashes', '=', itemHashes)" +
-      "])); " +
-      "items.filter(item => !processedItemHashesMap[xdmp.hash64(prefix + item)]);";
+    const script =
+       `const itemHashes ='${items}'.split(',').map(item => xdmp.hash64('${prefix}' + item)); ` +
```

Comment:
I think it would be safer to use 

let items = ${xdmp.toJsonString(items)};

And then use items as usual.